runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -275,8 +275,39 @@ ROMClassBuilder::handleAnonClassName(J9CfrClassFile *classfile, bool *isLambda,
 	BOOLEAN newCPEntry = TRUE;
 	BuildResult result = OK;
 	char buf[ROM_ADDRESS_LENGTH + 1] = {0};
+	U_8 *hostPackageName = context->hostPackageName();
+	UDATA hostPackageLength = context->hostPackageLength();
 	PORT_ACCESS_FROM_PORT(_portLibrary);
 
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	/* InjectedInvoker is a hidden class without the nestmate and strong
+	 * attributes set. It is created by MethodHandleImpl.makeInjectedInvoker on
+	 * the JCL side. ROM class name for InjectedInvoker is set using the hidden
+	 * class name, which contains the correct host class name. The below filter
+	 * is used to reduce the number of memcmps when identifying if a hidden
+	 * class is named InjectedInvoker.
+	 */
+	if (context->isClassHidden()
+	&& !context->isHiddenClassOptStrongSet()
+	&& !context->isHiddenClassOptNestmateSet()
+	) {
+#define J9_INJECTED_INVOKER_CLASSNAME "$$InjectedInvoker"
+		U_8 *nameData = context->className();
+		if (NULL != nameData) {
+			UDATA nameLength = context->classNameLength();
+			U_8 *start = nameData + nameLength - LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME);
```

Comment:
can you add brackets to make it more clear what is being added to the pointer. ie `nameData + (nameLength - LITERAL_STRLEN(J9_INJECTED_INVOKER_CLASSNAME));`